libs/agno/agno/agent/agent.py

```diff
@@ -877,21 +877,8 @@ def run(
 
         # When filters are passed manually
         if self.knowledge_filters or knowledge_filters:
-            """
-                initialize metadata (specially required in case when load is commented out)
-                when load is not called the reader's document_lists won't be called and metadata filters won't be initialized
-                so we need to call initialize_valid_filters to make sure the filters are initialized
-            """
-            if not self.knowledge.valid_metadata_filters:  # type: ignore
```

Comment:
This is the part that you've replaced with reading metadata from contents db right?